### PR TITLE
[106]: Set app locale to system locale on first launch

### PR DESCRIPTION
**Pull request by sanjeevyadavIT**
Changes:

1. Add `react-native-localize` library
2. When app first launch, use the system locale to set app locale if supported by app
3. Write new tests
4. Update `enUs` to `en-US` and `ptBR` to `pt-BR` everywhere

### CODE REVIEW

File: src/locales/__tests__/index.test.ts [hunk 1]
```diff
@@ -31,71 +32,100 @@ describe('Locales', () => {
     jest.clearAllMocks();
   });
 
-  // if don't has the locale in AsyncStorage, should set default value ptBR
-  test('caso não possua o locale no AsyncStorage, deve setar o default ptBR', async () => {
+  // if don't has the locale in AsyncStorage, should set default value from 'react-native-loalize'
+  test('caso não possua o locale no AsyncStorage, deve setar o default de react-native-loalize', async () => {
```
Comment (salomaoluiz) [inline src/locales/__tests__/index.test.ts:36]: here is just a grammar mistake, in Portuguese on this case, we use "... do react-native-localize'. And you wrote "loalize" here  and in the above line.

File: src/locales/__tests__/index.test.ts [hunk 2]
```diff
@@ -31,71 +32,100 @@ describe('Locales', () => {
...
+    expect(spyFindBestAvailableLanguage).toHaveBeenCalledWith(
+      getAvailableLocales(),
+    );
+    expect(i18n.locale).toEqual('en-US');
+
+    spyFindBestAvailableLanguage.mockRestore();
+  });
+
+  // if don't has the locale in AsyncStorage, and system locale is not supported, should set it to 'pt-BR'
+  test('caso não possua o locale no AsyncStorage, localidade do sistema não é suportado, deve configurá-lo como pt-BR', async () => {
```
Comment (salomaoluiz) [inline src/locales/__tests__/index.test.ts:57]: Here you can write something like this:
"caso não possua o locale no AsyncStorage e o locale não é suportado pelo sistema, deve configurá-lo como pt-BR"

**Review by salomaoluiz — CHANGES_REQUESTED** [review]
Hi @sanjeevyadavIT , thanks for your contributions, I just request two changes, it is littles mistakes on portuguese translation.

**Comment by sanjeevyadavIT** [comment]
I have made the requested changes

**Review by salomaoluiz — APPROVED** [review]

**Review by salomaoluiz — CHANGES_REQUESTED** [review]
Another thing, in the commit, add a # on issue ID, because this way are created a link on commit to issue, your commit will be this way: [#106]: Set app locale to system locale on first launch

**Comment by sanjeevyadavIT** [comment]
Hi, I have change the commit message

**Review by salomaoluiz — APPROVED** [review]

**Pull request merged**